src/layer/filter.rs

```diff
@@ -25,7 +23,7 @@ where
             self.layer.default_keyword
         };
 
-        if Mode::supports_enable_callback() {
+        if crate::native::Provider::<OutMode>::supports_enable_callback() {
```

Comment:
Using the exported type rather than trying to put everything into generics helped reduce the complexity a lot. The drawback is it makes hacking up a pseudo cross-compilation environment in VSCode harder.